Index: Source/Makefile
===================================================================
--- Source/Makefile	(revision 139)
+++ Source/Makefile	(working copy)
@@ -1,4 +1,5 @@
 TARGETS= Gen/float/libbox2d.a Gen/fixed/libbox2d.a
+TARGETS+= Gen/float/libbox2d.so Gen/fixed/libbox2d.so
 
 ifdef DEVKITPRO
 TARGETS += Gen/nds-float/lib/libbox2d.a Gen/nds-fixed/lib/libbox2d.a
@@ -71,6 +72,8 @@
 Gen/float/libbox2d.a:	$(addprefix Gen/float/,$(SOURCES:.cpp=.o))
 	ar cr $@ $^
 	ranlib $@ 
+Gen/float/libbox2d.so:	$(addprefix Gen/float/,$(SOURCES:.cpp=.o))
+	$(CXX) -shared -fPIC -C $^ -o $@ -lm -lgcc
 
 Gen/float/%.d:		%.cpp
 	@mkdir -p $(dir $@)
@@ -81,11 +84,12 @@
 	@mkdir -p $(dir $@)
 	c++ $(CXXFLAGS) -DTARGET_FLOAT32_IS_FIXED -c -o $@ $<
 
-
 Gen/fixed/libbox2d.a:	$(addprefix Gen/fixed/,$(SOURCES:.cpp=.o)) 
 	ar cr $@ $^
 	ranlib $@ 
 
+Gen/fixed/libbox2d.so:	$(addprefix Gen/fixed/,$(SOURCES:.cpp=.o)) 
+	$(CXX) -shared -fPIC -C $^ -o $@ -lm -lgcc
 
 Gen/fixed/%.d:		%.cpp
 	@mkdir -p $(dir $@)
@@ -129,6 +133,3 @@
 Gen/nds-float/%.d:		%.cpp
 	@mkdir -p $(dir $@) 
 	arm-eabi-g++ -MM -MT $(@:.d=.o) $(NDS_COMPILE_FLAGS) -o $@ $<
-
-
-
Index: Makefile
===================================================================
--- Makefile	(revision 139)
+++ Makefile	(working copy)
@@ -6,6 +6,25 @@
 	(cd Source; make)
 	(cd Examples/TestBed; make)
 
+install: all
+	install Source/Gen/float/libbox2d.a \
+	        Source/Gen/float/libbox2d.so \
+                /usr/local/lib
+	
+	for dir in Collision \
+	           Common \
+		   Dynamics \
+		   Collision/Shapes \
+		   Dynamics/Joints \
+		   Dynamics/Contacts; \
+        do \
+	  install -d /usr/local/include/box2d/Source/$$dir;\
+	  install Source/$$dir/*.h /usr/local/include/box2d/Source/$$dir/;\
+	done
+
+    #rewrite header file to make it possible to use it from the installed location.
+	cat Include/Box2D.h | sed s:\\.\\.:box2d: > /usr/local/include/box2d/Box2D.h
+
 clean:
 	(cd Contrib/freeglut; make clean)
 	(cd Contrib/glui; make clean)
Index: Examples/TestBed/Framework/Test.cpp
===================================================================
--- Examples/TestBed/Framework/Test.cpp	(revision 139)
+++ Examples/TestBed/Framework/Test.cpp	(working copy)
@@ -171,8 +171,8 @@
 		md.body2 = body;
 		md.target = p;
 #ifdef TARGET_FLOAT32_IS_FIXED
-		md.maxForce = (body->m_mass < 16.0)? 
-			(1000.0f * body->m_mass) : float32(16000.0);
+		md.maxForce = (body->GetMass() < 16.0)? 
+			(1000.0f * body->GetMass()) : float32(16000.0);
 #else
 		md.maxForce = 1000.0f * body->GetMass();
 #endif
